capirca/lib/cisco.py

```diff
@@ -640,6 +645,12 @@ def _FixOptions(self, proto, option):
     if ((proto == self.PROTO_MAP['udp'] or proto == 'udp')
         and 'established' in sane_options):
       sane_options.remove('established')
+    if ((proto == self.PROTO_MAP['tcp'] or proto == 'tcp')
```

Comment:
I might be incorrect here as I have not worked much on this section of the code but I believe this is used to prevent the printing of the word established in cases where a udp established term would be used. The established field is useful in some cases. If fragments are not valid to be used with UDP or TCP it is likely that instead of filtering we should be marking these invalid by erroring out with a message.

Reply:
This whole method "_FixOptions", from my point of view it doesn't make sense. If an option is invalid, I totally agree with you that it should be pointed instead of just "fixing under the hood". Indeed, imho this is a bug. If someone use the tcp-established option with udp protocol, capirca shouldn't create the ACL. For Cisco.py, there were only two valid sub-tokens for token option: "tcp-established" and "established". The decision done when this was implemented was to avoid checking this options but fixing any wrong option used, so I followed the same principle. I've just changed that and removed that part. But then, if an option is wrongly used, this option is removed without any notification (this is how Capirca works). I could create a new PR to improve this.

Reply:
sgtm